### PR TITLE
Add Balancer Gauge Labels

### PR DESCRIPTION
**Pull request by hedgehog-jacek**
Brief comments on the purpose of your changes:

 - Added Balancer v2 gauge labels.
 
 cc: @markusbkoch @Stefenon @mendesfabio 

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] ~~`coin_id` represents the ID of the coin on coinpaprika.com~~
* [ ] ~~all the coins are active on coinpaprika.com (please remove inactive ones)~~

### Join logic:
* [ ] ~~if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible~~

### Incremental logic:
* [ ] ~~I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables~~
  * [ ] ~~where block_time >= date_trunc("day", now() - interval '1 week')~~
* [ ] ~~if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')~~
* [ ] ~~if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')~~

### CODE REVIEW

**Review by jeff-dude — CHANGES_REQUESTED** [review]
thank you for the PR @hedgehog-jacek

some teammates / wizards recently redesigned labels, please see:
https://github.com/duneanalytics/spellbook/discussions/2634
https://github.com/duneanalytics/spellbook/pull/2680

it'd be good if you can try to find the pieces to update in this PR. one example is the blockchain column, which is no longer an array, but just a string. that's one simple one. maybe even @andrewhong5297 can help review this :)

**Comment by jeff-dude** [comment]
> thank you for the PR @hedgehog-jacek
> 
> some teammates / wizards recently redesigned labels, please see: #2634 #2680
> 
> it'd be good if you can try to find the pieces to update in this PR. one example is the blockchain column, which is no longer an array, but just a string. that's one simple one. maybe even @andrewhong5297 can help review this :)

there's also a new readme in the linked PR, ideally with some more help

File: models/labels/balancer_v2/labels_balancer_v2_gauges_arbitrum.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{{config(alias='balancer_v2_gauges_arbitrum',
+        post_hook='{{ expose_spells(\'["arbitrum"]\',
+                                    "sector",
+                                    "labels",
+                                    \'["jacektrocinski"]\') }}')}}
+
+SELECT
+    array('arbitrum') AS blockchain,
```
Comment (andrewhong5297) [inline models/labels/balancer_v2/labels_balancer_v2_gauges_arbitrum.sql:8]: we're no longer using `array<string>` as the blockchain column, just a `string` now.

File: models/labels/balancer_v2/labels_balancer_v2_gauges_arbitrum.sql [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    gauge.gauge AS address,
+    'arb:' || pools.name AS name,
+    'balancer_v2_gauges' AS category,
+    'balancerlabs' AS contributor,
+    'query' AS source,
+    TIMESTAMP('2022-01-13') AS created_at,
+    NOW() AS updated_at
```
Comment (andrewhong5297) [inline models/labels/balancer_v2/labels_balancer_v2_gauges_arbitrum.sql:15]: you'll need to add two lines for `model_name` which should just be `balancer_v2_gauges` and `label_type` as `identifier`

File: models/labels/balancer_v2/labels_balancer_v2_schema.yml [hunk 3]
```diff
@@ -9,36 +9,36 @@ models:
...
-        description: "How were labels generated (could be static or query)"
+        description: 'How were labels generated (could be static or query)'
       - &created_at
         name: created_at
-        description: "When were labels created"
+        description: 'When were labels created'
       - &updated_at
         name: updated_at
-        description: "When were labels updated for the last time"
+        description: 'When were labels updated for the last time'
```
Comment (andrewhong5297) [inline models/labels/balancer_v2/labels_balancer_v2_schema.yml:42]: see other schemas after you merge from main for `model_name` and `label_type` column descriptors. Add to all of the below schemas too.

**Comment by hedgehog-jacek** [comment]
@Stefenon can you work on this PR?

**Comment by Stefenon** [comment]
> @Stefenon can you work on this PR?

Sure, I'll work on it!

**Comment by Stefenon** [comment]
Hey @andrewhong5297!

I have worked in your suggestions for this PR, but after I merged main into it, the tests are failing. I took a look at it and apparently it is a problem with the columns combination uniqueness.

I [queried]() all the labels where these columns repeat and the number of results is the same as the described in the [test error](https://github.com/duneanalytics/spellbook/actions/runs/4305103998/jobs/7507067414#step:10:38), but none of them are related to Balancer labels.

Do you have any suggestions on how to move on with this PR to merge it?

**Comment by jeff-dude** [comment]
> Hey @andrewhong5297!
> 
> I have worked in your suggestions for this PR, but after I merged main into it, the tests are failing. I took a look at it and apparently it is a problem with the columns combination uniqueness.
> 
> I [queried](https://dune.com/queries/2097356) all the labels where these columns repeat and the number of results is the same as the described in the [test error](https://github.com/duneanalytics/spellbook/actions/runs/4305103998/jobs/7507067414#step:10:38), but none of them are related to Balancer labels.
> 
> Do you have any suggestions on how to move on with this PR to merge it?

following up with @andrewhong5297 to see if we can progress this PR

**Comment by andrewhong5297** [comment]
> > Hey @andrewhong5297!
> 
> > 
> 
> > I have worked in your suggestions for this PR, but after I merged main into it, the tests are failing. I took a look at it and apparently it is a problem with the columns combination uniqueness.
> 
> > 
> 
> > I [queried](https://dune.com/queries/2097356) all the labels where these columns repeat and the number of results is the same as the described in the [test error](https://github.com/duneanalytics/spellbook/actions/runs/4305103998/jobs/7507067414#step:10:38), but none of them are related to Balancer labels.
> 
> > 
> 
> > Do you have any suggestions on how to move on with this PR to merge it?
> 
> 
> 
> following up with @andrewhong5297 to see if we can progress this PR

Yeah the labels.all file ends up with uniqueness issues due to wrong logic in other peoples files. I'm going to force fix some of these this week so this can be unblocked

**Comment by jeff-dude** [comment]
closing due to inactivity

**Comment by jeff-dude** [comment]
> Hey @jeff-dude, I didn't realize this PR was blocking from our side, I'm sorry. I believe this should run fine if we merge main, should I do it so that we can reopen the PR?

definitely feel free to reopen and merge main/fix any conflicts -- happy to have it active again

**Pull request closed without merging**